local-tests/chainrunner/node.py

```diff
@@ -0,0 +1,92 @@
+import json
+import os.path as op
+import re
+import requests
+import jsonrpcclient as rpc
+import subprocess
+
+from .utils import flags_from_dict
+
+class node:
+    """A class representing a single node of a running blockchain."""
+    def __init__(self, binary, chainspec, path, logdir=None):
+        self.chainspec = chainspec
+        self.binary = binary
+        self.path = path
+        self.logdir = logdir or path
+        self.logfile = None
+        self.flags = {}
+        self.running = False
+
+
+    def _stdargs(self):
+        return ['--base-path', self.path, '--chain', self.chainspec]
+
+
+    def start(self, name):
+        """Start the node. `name` is used to name of the logfile and for --name flag."""
+        cmd = [self.binary, '--name', name] + self._stdargs() + flags_from_dict(self.flags)
+        self.logfile = op.join(self.logdir, name+'.log')
+        self.process = subprocess.Popen(cmd, stderr=open(self.logfile, 'w'), stdout=subprocess.DEVNULL)
+        self.running = True
+
+
+    def stop(self):
+        """Stop the node by sending SIGKILL."""
+        self.process.kill()
+        self.running = False
+
+
+    def purge(self):
+        """Purge chain (delete the database of the node)."""
+        cmd = [self.binary, 'purge-chain', '-y'] + self._stdargs()
+        subprocess.run(cmd, stdout=subprocess.DEVNULL).check_returncode()
+
+
+    def greplog(self, regexp):
```

Comment:
matter of taste, but I would pass here already a regex object (thus good to have a type hint :P) to avoid compiling arbitrary string within this method - it's responsibility is to search through logs, so it is a little bit confusing if it fails because the passed regex couldn't been compiled.

Reply:
For interactive use (and I would guess this might be a common use case of that method), especially IPython console, it's much more convenient to be able to quickly grep for some string with oneliner:
`chain[2].greplog('ERROR')`
rather than:
```import re
x = re.compile('ERROR')
`chain[2].greplog(x)```

Reply:
okay, so use `re.findall()` without explicit compiling

Reply:
Ok